src/webapi/xml_http_request.rs

```diff
@@ -91,6 +91,31 @@ impl XMLHttpRequest {
         };
     }
 
+    /// Returns the string containing the text of the specified header. If there
+    /// are multiple response headers with the same name, then their values are
+    /// returned as a single concatenated string.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/getRequestHeader)
+    pub fn get_request_header(&self, header: &str) -> Option<String> {
+        let header = js!( return @{self}.getRequestHeader(@{header}); );
+        match header {
+            Value::Null => None,
+            Value::String(text) => Some(text),
+            _ => unreachable!(),
+        }
+    }
+
+    /// Sets the value of an HTTP request header. Must be called after `open()`,
+    /// but before `send()`. If this method is called several times with the same
+    /// header, the values are merged into one single request header.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/setRequestHeader)
+    pub fn set_request_header(&self, header: &str, value: &str) {
+        js! {
```

Comment:
I forgot this in the last review, but if you have a `js!` macro invocation from which you don't need to return anything then please add `@(no_return)` at the start. It will be marginally more efficient and generate less code.